src/azul/plugins/metadata/hca/transform.py

```diff
@@ -940,6 +943,31 @@ def _sequencing_input(self, sequencing_input: api.Biomaterial) -> MutableJSON:
             'sequencing_input_type': api.schema_names[type(sequencing_input)]
         }
 
+    @classmethod
+    def _aggregate_date_types(cls) -> FieldTypes:
+        return {
+            'document_id': null_str,
+            'submission_date': null_datetime,
+            'update_date': null_datetime,
+        }
+
+    def _aggregate_date(self) -> MutableJSON:
+        submission_dates = (
+            entity.submission_date
+            for entity in self.api_bundle.entities.values()
+            if not entity.is_stitched
+        )
+        update_dates = (
+            entity.update_date
+            for entity in self.api_bundle.entities.values()
+            if not entity.is_stitched
+        )
+        return {
+            'document_id': str(self.api_bundle.uuid),
+            'submission_date': min(submission_dates, key=none_safe_key()),
```

Comment:
If `None` is in the iterable, `min` would pick it. If you end up having to write a generator expression, it might be better to use a conditional instead of none_safe_key. Otherwise use `none_last=True`.